entity-framework/core/managing-schemas/scaffolding.md

```diff
@@ -1,14 +1,138 @@
+
+## Directories and namespaces
+
+The entity classes and a DbContext class are scaffolded into the project's root directory and use the project's default namespace. You can specify the directory where classes are scaffolded using `-OutputDir` (PMC) or `--output-dir` (CLI). The namespace will be the root namespace plus the names of any subdirectories under the project's root directory.
+
+You can also use `-ContextDir` (PMC) and `--context-dir` (CLI) to scaffold the DbContext class into a separate directory from the entity type classes.
+
+``` powershell
+Scaffold-DbContext ... -ContextDir Data -OutputDir Models
+```
+
+``` Console
+dotnet ef dbcontext scaffold ... --context-dir Data --output-dir Models
+```
+
+## How it works
+
+Reverse engineering starts by reading the database schema. This includes information about tables, columns, constraints, and indexes.
+
+Next, it uses the schema information to create an EF Core model. Tables are used to create entity types; columns are used to create properties; and foreign keys are used to create relationships.
+
+Finally, the model is used to generate code. The corresponding entity type classes, Fluent API, and data annotations are scaffolded in order to re-create the same model from your app.
+
+## What doesn't work
+
+Not everything about a model can be represented using a database schema. For example, information about **inheritance hierarchies**, **owned types**, and **table splitting** are not present in the database schema. Because of this, these constructs will never be reverse engineered.
+
+In addition, **some column types** may not be supported by the EF provider. These columns won't be included in the model.
+
+EF Core requires every entity type to have a key. Tables, however, aren't required to specify a primary key. **Tables without a primary key** are currently not reverse engineered.
+
+You can define **concurrency tokens** in an EF model to prevent two users from updating the same entity at the same time. Some databases have a special type to represent this type of column (e.g. rowversion in SQL Server) in which case we can reverse engineer this information; however, other concurrency tokens will not be reverse engineered.
+
+## Customizing the model
+
+After the entity type classes and DbContext class are scaffolded, you are free to customize them to fit your needs. For example, you may choose to rename types and properties, introduce inheritance hierarchies, or split a table into to multiple entities. You can also remove non-unique indexes, unused sequences and navigation properties, optional scalar properties, and constraint names from the model. The scaffolded code represents *one* model that can be used to access the database, but it's certainly not the *only* model that can be used.
+
+> [!WARNING]
+> If you reverse engineer the model from the database again, any customization you make to the files will be overwritten.
+
+## Updating the model (re-scaffolding)
```

Comment:
The topics "updating the model (re-scaffolding)" and "customizing the model" seem to be closely coupled. E.g. it seems it would be more understandable if we mentioned that we generate partial classes and how you can take advantage of that in "customizing the model". But it also seems to make sense to mention in the warning you put in "customizing the model" that you have to use `--force` for the files to be overwritten. 

Perhaps shuffling the contents around and making this a single title would help.